tests/testsuite/bad_config.rs

```diff
@@ -781,7 +792,26 @@ fn invalid_toml_historically_allowed_fails() {
 
     p.cargo("build")
         .with_status(101)
-        .with_stderr_contains("  expected newline, found an identifier at line 1 column 7")
+        .with_stderr(
+            "\
+error: could not load Cargo configuration
+
+Caused by:
+  could not parse TOML configuration in `[..]`
+
+Caused by:
+  could not parse input as TOML
+
+Caused by:
+  TOML parse error at line 1, column 7
+    |
+  1 | [bar] baz = 2
+    |       ^
+  Unexpected `b`
+  Expected `newline` or `end of input`
```

Comment:
Nit: "newline" and "end of input" shouldn't be in backquotes; only literal syntax should be in backquotes. This should say "Expected newline or end of input".

Reply:
I agree with these suggestions, the question comes down to prioritization.  Almost all of the error formatting is coming from the underlying `combine` crate.  To change it, we have to move away from their "easy" error handling which would take some work.  I know this specific one was a `nit` but not all of them were written as such.  So the question is how important are addressing these for cargo short term (including blocking this PR) and long term (compared to cargo-add or many other potential projects).

Reply:
@epage I don't think these changes need to block this PR, as long as they have corresponding open issues on `toml_edit`.

There's only one item below I'd like further clarification on before this PR goes in. I'll follow up on that one.

Reply:
https://github.com/ordian/toml_edit/issues/295

Reply:
I've also made sure everything in my [earlier post](https://github.com/rust-lang/cargo/pull/10086#issuecomment-1003418045) has an issue and added the cases you called out, with issues.

Reply:
Thank you!